### DISCUSSION TITLE
Is there systemd unit state source?

### DISCUSSION THREAD

**Discussion by Javex** [vectordotdev/vector#20581] [Q&A] [answered]
I just discovered vector and am contemplating whether to move monitoring to it instead of fluent-bit. Currently I use promtail & node_exporter, but I'm looking for something new and was playing around with fluent-bit when I came across vector. One important monitor for me is the `node_systemd_unit_state` metric provided by node_exporter (and which fluent-bit also supports as a drop-in replacement).

For vector, I noticed [host metrics](https://vector.dev/docs/reference/configuration/sources/host_metrics/) exist, but they don't seem to support systemd metrics and I can't find any metrics source for systemd, only journald for logs. Am I missing something? Does this source exist? If not, am I even asking the right question? How would you find out if a systemd unit has failed with Vector?

Thanks!

**Comment by jszwedko** [comment, marked as answer]
Hi @Javex !

Unfortunately that metric doesn't exist in Vector at the moment. I think your best bet would be to run `node_exporter` alongside Vector and have Vector scrape it using the `prometheus_scrape` source. There was some discussion about the lack of node exporter metrics in Vector here: https://github.com/vectordotdev/vector/issues/17734

**Reply by Javex** [reply]
Thank you! After reading through the linked issues it seems my best choice is to stick with node exporters or use fluent bit since that fits my use case. Cheers!
